File: frontend/lib/models/language.dart
class Language {
  final String id;
  final String name;
  final String proficiency;

  Language({
    required this.id,
    required this.name,
    required this.proficiency,
  });

  factory Language.fromJson(Map<String, dynamic> json) {
    return Language(
      id: json['_id'],
      name: json['name'],
      proficiency: json['proficiency'],
    );
  }
}

File: backend/models/Language.js
const mongoose = require('mongoose');

const LanguageSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        unique: true
    },
    proficiency: {
        type: String,
        required: true,
        enum: ['Beginner', 'Intermediate', 'Fluent', 'Native']
    },
    createdAt: {
        type: Date,
        default: Date.now
    }
});

module.exports = mongoose.model('Language', LanguageSchema);
